packages/charts/examples/components/demo/advanced-usage/legend/legend-metadata/legend-metadata-example.component.ts

```diff
@@ -0,0 +1,156 @@
+        const accessors =  stackedAreaAccessors();
+        // 'y1' defines access to the numeric values we want to visualize.
+        // The items in the data array of this example have a property named 'value',so we'll use that.
+        accessors.data.y1 = (d) => d.value;
+
+        // The area renderer will make the chart look like a area chart.
+        const renderer = new AreaRenderer();
+
+        // In case of a area chart, the scale definitions are flexible.
+        // This example demonstrates a scenario with time on the X scale and a numeric value on the Y scale.
+        const scales: IXYScales = {
+            x: new TimeScale(),
+            y: new LinearScale(),
+        };
+
+        const averageData = getAverage(getData());
+        // We are using a different renderer so the metadata does not effect the domain
+        const noopRenderer = new XYRenderer({ignoreForDomainCalculation: true});
+        // Here we create an accessor for our average metadata
+        const avgAccessors = new XYAccessors();
+        // This is so the legend knows the value for the y
+        avgAccessors.data.y = (d) => d.value;
+        // Same as above we are setting the numeric value we want to visualize
+        avgAccessors.data.y1 = avgAccessors.data.y;
+        // Setting the color of the series to transparent so it doesn't render
+        avgAccessors.series.color = () => "var(--nui-color-bg-transparent)";
+        // Deleting the markers so it doesn't place a marker on the chart
+        delete avgAccessors.series.marker;
+
+        this.avgSeries = {
+            ...averageData,
+            accessors: avgAccessors,
+            renderer: noopRenderer,
+            scales: scales,
+            showInLegend: false,
+            preprocess: false,
+        }
+        // Here we assemble the complete chart series.
+        let seriesSet: Partial<IChartSeries<IAccessors>>[] = getData().map(d => ({
+            ...d,
+            accessors,
+            renderer,
+            scales,
+        }));
+        // Combining both sets in an array of series
+        seriesSet = [...seriesSet, this.avgSeries];
+
+        // Finally, pass the series set to the chart's update method
+        this.chartAssist.update(seriesSet as IChartSeries<IAreaAccessors>[]);
+    }
+}
+
+function getAverage(dataArr: any) {
+    let arrAverage = [];
+    const dataLength =  dataArr[0].data.length;
+    const numOfSeries = dataArr.length
+    for (let n = 0; n < dataLength; n++) {
+        let avg = 0;
+        for (let i = 0; i < dataArr.length; i++) {
+            const series = dataArr[i];
+            avg += series.data[n].value;
+        }
+        avg = avg/numOfSeries
+        arrAverage.push({ x: dataArr[0].data[n].x, value: avg });
+    }
+    return {
+        id: "average",
+        name: "Average Speed",
+        data: arrAverage,
+    };
+}
+
+/* Chart data */
+function getData() {
```

Comment:
Maybe a return type:

```suggestion
function getData(): Partial<IDataSeries<XYAccessors>>[] {
```